Allow User to Subscribe, and Display a List of Public Repos

yakschuss opened the pull request.

yakschuss committed:
Add Homepage Styling/Nav

I prepared the homepage for OmniAuth Github Functionality. Gems, Nav,
and Styles were added.

yakschuss committed:
Add Devise and User Model

Added Devise to interact with Github OmniAuth, allow a user to sign in
and give WebHook access to public repos.

yakschuss committed:
Grab User Repo from Github

When a User Authorizes our Application, we get the repos from the public
github profile, and create Repo objects for each one. We then populate
them on Dashboard#show.

Also, some rubocop fixes and edits.

yakschuss committed:
Fix RuboCop and SCSS-Lint Errors

Added a color variable file, and ordered rules properly
Cleaned up all Rubocop Errors

alexstophel commented on db/migrate/20161030225939_create_repos.rb:
```diff
@@ -0,0 +1,11 @@
+class CreateRepos < ActiveRecord::Migration[5.0]
+  def change
+    create_table :repos do |t|
```
I would add `null: false` on these columns.

alexstophel replied:
You'll also want to have matching validations in the models for any column you add `null: false` to.

alexstophel commented on db/migrate/20161030141745_add_user_info_to_user.rb:
```diff
@@ -0,0 +1,6 @@
+class AddUserInfoToUser < ActiveRecord::Migration[5.0]
+  def change
+    add_column :users, :username, :string
```
If you're going to rely on a user having a username, it's best to have `null: false` on this column.

alexstophel commented on lib/github_client.rb:
```diff
@@ -0,0 +1,30 @@
+
+  private
+
+  attr_accessor :user
```
I would make this an `attr_reader` unless you plan on allowing on allowing something in this class to change `@user` later.

alexstophel commented on lib/github_client.rb:
```diff
@@ -0,0 +1,30 @@
+  end
+
+  def client
+    @client ||= client.new(access_token: ENV["ACCESS_TOKEN"])
```
I prefer to memoize by prefixing an underscore. It makes it clear that this should not be changed. `@_client`

yakschuss committed:
Add Validations To Models

Added Null:False to model attributes, matching validations.
Also fixed some style issues and failing tests.
